### DISCUSSION TITLE
Timer update charge period

### DISCUSSION THREAD

**Discussion by ITSpecialist111** [nathanmarlor/foxess_modbus#401] [General]
Firstly this is an amazing intergration!! 

I've found a bug that if you're trying to set the Force charge and charge from grid to 'Off', but also want to update the timer to 00:00:00 this is impossible.

Work around:  To have the period "off" the change requires at least 'force charge' enabled, and to set the period start & End a minute apart.

You can't have this time overlap with another.
You can't set 00:00:00 on each start & end   -   Fox removed the force change option in the UI. And now relies on the timer to be zero'ed to be ignored. (If the timer is set it will now act like the force charge switch).

![image](https://github.com/nathanmarlor/foxess_modbus/assets/142685548/e92a657c-41c7-40f9-b38d-7a542ba1dec6)

**Comment by canton7** [comment]
I really need to write an faq on this... This is the intended behaviour. To disable a charge period, disable the "Enable force charge" toggle. It will send 00:00-00:00 to the inverter and remember the last times you had set.

**Reply by ITSpecialist111** [reply]
In my testing that doesn't have an effect on the timer. It just leaves the previous start and end times.

**Reply by canton7** [reply]
It sends 00:00-00:00 to the inverter *and remembers the previous times you set*

**Reply by canton7** [reply]
This is for the charge period card, so you can re-enable the charge period without having to re-enter the times.

**Reply by ITSpecialist111** [reply]
MMmm I'm not using the charge period card. If you already have the timers set as:  01:00:00 and 03:00:00   and turn off the binary sensors. the inverter ignores the time changes


alias: Check Octopus Intelligent Dispatch Slot OFF and Battery SOC
trigger:
  - platform: time_pattern
    minutes: /2
condition:
  - condition: state
    entity_id: binary_sensor.octopus_intelligent_planned_dispatch_slot
    state: "off"
action:
  - service: foxess_modbus.update_charge_period
    data:
      inverter: 73afbbec6a8de5343770d3d01e40ff67
      enable_force_charge: false
      enable_charge_from_grid: false
      start: "00:00:00"
      end: "00:00:00"
      charge_period: "2"

**Reply by canton7** [reply]
> the inverter ignores the time changes

What do you mean by that?

Please see the conversation here: https://m.facebook.com/groups/foxessownersgroup/permalink/1917321598650013/?paipv=0&eav=AfbwNQriwVl0L1u3pONnBObprkPrC2klWzXJRBDKjf-ooxpoVneCxGaO3wvzbYyiF1k&wtsid=rdr_0uazARHvMlFJgFimT&_rdr

**Reply by canton7** [reply]
See also https://github.com/nathanmarlor/foxess_modbus/issues/367 and https://github.com/nathanmarlor/foxess_modbus/issues/146

**Reply by canton7** [reply]
If your automation ends up calling the service, it will send 00:00-00:00 to the inverter to disable the charge period. The integration however will remember the last-set times, and will display them. You can check what's set on the inverter using the front panel. Note that the foxess app seems to show changes to the charge period straight away (as long as you exit/re-enter the appropriate page), but if you disable a charge period it can take half an hour to show on the app. That is a cloud thing: it takes the inverter a while to send updates to the cloud.
